config/worker.js

```diff
@@ -0,0 +1,146 @@
+            return new Response(null, {
+                status: 302,
+                headers: {
+                    'location': url
+                }
+            })
+        }
+        case '/refreshtoken': {
+            let code = request.headers.get('Authorization')
+            if(request.method == 'OPTIONS') {
+                return new Response(null, {
+                    status: 204,
+                    headers: {
+                        'Access-Control-Allow-Origin': '*',
+                        'Access-Control-Allow-Methods': 'GET',
+                        'Access-Control-Allow-Headers': 'Authorization'
+                    }
+                })
+            }
+            let resp = await fetch('https://accounts.spotify.com/api/token', {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/x-www-form-urlencoded',
+                    'Authorization': 'Basic ' + btoa(`${CLIENT_ID}:${CLIENT_SECRET}`) //clientID:clientSecret in base64
+                },
+                body: `grant_type=authorization_code&code=${code}&redirect_uri=${encodeURIComponent(CALLBACK_URL)}`
+            }).then((spotiRes) => spotiRes.json())
+            .then(body => {
+                if(body.error) {
+                    return new Response(JSON.stringify(body), {
+                        status: 502,
+                        headers: {
+                            'Access-Control-Allow-Origin': '*',
+                        }
+                    })
+                }
+                return new Response(body.refresh_token, {
+                    status: 200,
+                    headers: {
+                        'Access-Control-Allow-Origin': '*',
+                    }
+                })
+            }).catch(e => {
+                console.log(e)
+                return new Response(e, {
+                    status: 502,
+                    headers: {
+                        'Access-Control-Allow-Origin': '*',
+                    }
+                })
+            })
+            return resp;
+        }
+        case '/accesstoken': {
+            let authCode = request.headers.get('Authorization')
+            if(request.method == 'OPTIONS') {
+                return new Response(null, {
+                    status: 204,
+                    headers: {
+                        'Access-Control-Allow-Origin': '*',
+                        'Access-Control-Allow-Methods': 'GET',
+                        'Access-Control-Allow-Headers': 'Authorization'
+                    }
+                })
+            }
+            if(!authCode) {
+                return new Response('woop', {
+                    status: 401,
+                    headers: {
+                        'Access-Control-Allow-Origin': '*'
+                    }
+                })
+            } else {
+                let resp = await fetch('https://accounts.spotify.com/api/token', {
```

Comment:
Told you already, you shouldn't be hardcoding the API url. It should be in the config file.